aarch64: build: update Compute Library min. version to v21.05

nSircombe opened the pull request:
# Description                                                                                                                                                                                                                                                                   

Updates minimum Compute Library version for builds on AArch64 with the
Compute Library backend. Updates from 21.02 to 21.05

- DroneCI build updated
- Docs updated
- CMake updated


# Checklist

## Code-change submissions

- [X] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally?
- [N/A] Have you formatted the code using clang-format?

### New features

- [N/A] Have you added relevant tests?
- [N/A] Have you provided motivation for adding a new feature?

### Bug fixes

- [N/A] Have you added relevant regression tests?
- [N/A] Have you included information on how to reproduce the issue (either in a github issue or in this PR)?

nSircombe commented:
Would it be possible to add this change to the 2.3 release branch too, if I raise another PR against it?

vpirogov commented:
Sure, please open the PR against rls-v2.3.

igorsafo commented:
Hi @nSircombe , what is the reason for increasing the minimal supported version? Since there are no changes in the code base I assume that oneDNN 2.3 will work with 21.03. However I understand that we might want to increase version in testing scripts to test oneDNN against the latest ACL version.

Regards,
Igor Safonov

nSircombe commented:
Hi @igorsafo,

Ideally I'd have like this change to be introduced when we updated the thread binding (#1063) and fixed a minor issue with the Scheduler (#1045), but the release timescales didn't workout. 

There are improvements in 21.05 to the thread spin-up at high thread counts which, along side the changes to Compute Library's thread binding, have a positive impact on performance. We'd like to make sure users get this 'by default' - that's the rationale for asking about inclusion in the 2.3 Release branch.

But there are also some known issues (such as the use of scratch pad memory to store biases (https://github.com/oneapi-src/oneDNN/blob/8715d47411778baaaaf1472092fe80e88b6a4a32/src/cpu/aarch64/acl_gemm_convolution.hpp#L126)) which we will be able to remove with 21.05 as the 'default', as well as some additional capability we're working on (e.g. a Compute Library based inner-product primitive) which depend on 21.05. Adding this change ahead of those, while not a necessity, does make life a little easier too.

nSircombe commented:
@vpirogov 

PR raised against rls-2.3 #1083 

...appologies about #1082, my bad.

igorsafo commented:
@nSircombe Thanks for the explanation, it makes sense now.

dzarukin reviewed (approved):
LGTM, thanks.
@nSircombe, quick question, do we want to re-trigger the testing for this PR or, given that it worked fine for v2.3, just hope it would work well, too?

nSircombe commented:
Hi @dzarukin,

> do we want to re-trigger the testing for this PR

...I'm not really sure what's happened there, it looks like all the failures are due to `git fetch` failures, but the DroneCI build appears happy!

I've pushed a `git commit --amend --no-edit` which appears to have re-triggered the builds, lets see what happens this time!

dzarukin commented:
Thank you for the contribution.
